examples/api/lib/material/autocomplete/autocomplete.5.dart

```diff
@@ -0,0 +1,249 @@
+                  'Type below to autocomplete the following possible results: ${_FakeAPI._kOptions}.'),
+              const _AsyncAutocomplete(),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _AsyncAutocomplete extends StatefulWidget {
+  const _AsyncAutocomplete();
+
+  @override
+  State<_AsyncAutocomplete> createState() => _AsyncAutocompleteState();
+}
+
+class _AsyncAutocompleteState extends State<_AsyncAutocomplete> {
+  // The query currently being searched for. If null, there is no pending
+  // request.
+  String? _currentQuery;
+
+  // The most recent options received from the API.
+  late Iterable<String> _lastOptions = <String>[];
+
+  late final _Debounceable<Iterable<String>?, String> _debouncedSearch;
+
+  // Check if the "remote" API's response is waiting to be received.
+  bool _isLoading = false;
+
+  String _lastValue = '';
+  bool _isFirstFocus = false;
+  bool _nothingFound = false;
+
+  // Calls the "remote" API to search with the given query. Returns null when
+  // the call has been made obsolete.
+  Future<Iterable<String>?> _search(String query) async {
+    _currentQuery = query;
+
+    // In a real application, there should be some error handling here.
+    final Iterable<String> options = await _FakeAPI.search(_currentQuery!);
+
+    // If another search happened after this one, throw away these options.
+    if (_currentQuery != query) {
+      return null;
+    }
+    _currentQuery = null;
+
+    return options;
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _debouncedSearch = _debounce<Iterable<String>?, String>(_search);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Autocomplete<String>(
+      showOptionsViewOnUncompletedOptions: true,
+      showOptionsViewOnEmptyOptions: true,
+      optionsBuilder: (TextEditingValue textEditingValue) async {
+        _isFirstFocus = textEditingValue.text.isEmpty;
+        _nothingFound = false;
+        if (_isFirstFocus) {
+          return const Iterable<String>.empty();
+        }
+        _lastValue = textEditingValue.text;
+        _isLoading = true; // Begin options loading.
+
+        final Iterable<String>? options =
+            await _debouncedSearch(textEditingValue.text);
+
+        if (textEditingValue.text == _lastValue) {
+          _isLoading = false; // End options loading.
+        }
+        if (options == null) {
+          return _lastOptions;
+        }
+        _lastOptions = options;
+        _nothingFound = options.isEmpty;
+        return options;
+      },
+      onSelected: (String selection) {
+        debugPrint('You just selected $selection');
+      },
+      optionsViewBuilder: (BuildContext context,
+          AutocompleteOnSelected<String> onSelected, Iterable<String> options) {
+        return _isLoading
+            ? const Text('loading...')
+            : _isFirstFocus
+                ? const Text('Type something')
+                : _nothingFound
```

Comment:
I think you can get rid of this variable and just use `options.isEmpty`, right?

Reply:
Yes.